src/radical/pilot/task_description.py

```diff
@@ -80,12 +97,85 @@ class TaskDescription(ru.Description):
        attribute can be used to map individual tasks back to application level
        workloads.
 
+    .. data:: mode
+
+       [type: `str` | default: `"executable"`] The execution mode to be used for
+       this task.  The following modes are accepted:
+
+         - EXECUTABLE: the task is spawned as an external executable via a
+           resource specific launch method (srun, aprun, mpiexec, etc).
+           required attributes: `executable`
+           related  attributes: `arguments`
+
+         - FUNCTION: the task references a python function to be called.
+           required attributes: `function`
+           related  attributes: `args`
+           related  attributes: `kwargs`
+
+         - EVAL: the task is a code snippet to be evaluated.
+           required attributes: `code`
+
+         - EXEC: the task is a code snippet to be `exec`'ed.
+           required attributes: `code`
+
+         - SHELL: the task is a shell command line to be run.
+           required attributes: `command`
+
+         - PROC: the task is a single core process to be executed.
+           required attributes: `executable`
+           related  attributes: `arguments`
+
+        There exists a certain overlap between `EXECUTABLE`, `SHELL` and `PROC`
+        modes.  As a general rule, `SHELL` and `PROC` should be used for short
+        running tasks which require a single core and no additional resources
+        (gpus, storage, memory).  `EXECUTABLE` should be used for all other
+        tasks and is in fact the default.  `SHELL` should only be used if the
+        command to be run requires shell specific functionality (pipes, I/O
+        redirection) which cannot easily be mapped to other task attributes.
+
+
     .. data:: executable
 
        [type: `str` | default: `""`] The executable to launch. The executable
        is expected to be either available via `$PATH` on the target resource,
        or to be an absolute path.
 
+    .. data:: arguments
+
+       [type: `list` | default: `[]`] The command line arguments for the given
+       `executable` (`list` of `strings`).
+
+    .. data:: code
+
+       [type: `str` | default: `""`] The code to run.  This field is expected to
+       contain valid python code which is executed when the task mode is
+       `code:exec` or `code:eval`.
```

Comment:
keep modes just as `exec` or `eval`?

Reply:
Ah, thanks - that is a left over from an earlier iteration - will clean up!